src/utils/entity_utils/inc/aduc/auto_file_entity.hpp

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file auto_file_entity.h
+ * @brief Defines AutoFileEntity class that houses and manages an ADUC_FileEntity member.
+ *
+ * @copyright Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+#ifndef AUTO_FILE_ENTITY
+#define AUTO_FILE_ENTITY
+
+#include <aduc/parser_utils.h> // ADUC_FileEntity_Uninit
+#include <aduc/types/update_content.h> // ADUC_FileEntity
+#include <cstring> // memset
+
+class AutoFileEntity
+{
+    ADUC_FileEntity entity{};
+
+public:
+    AutoFileEntity(const AutoFileEntity&) = delete;
+    AutoFileEntity& operator=(const AutoFileEntity&) = delete;
+    AutoFileEntity(AutoFileEntity&&) = delete;
+    AutoFileEntity& operator=(AutoFileEntity&&) = delete;
+
+    AutoFileEntity() = default;
+
+    ~AutoFileEntity()
+    {
+        uninit();
+    }
+
+    const ADUC_FileEntity* operator->() const
+    {
+        return &entity;
+    }
+
+    /**
+     * @brief Get the const addres of the file entity object
+     * @return const ADUC_FileEntity* Address of wrapped object.
+     */
+    const ADUC_FileEntity* get() const
+    {
+        return &entity;
+    }
+
+    /**
+     * @brief Get the address of the file entity object.
+     * Useful for passing to methods with the signature "f(..., ADUC_FileEntity* outFileEntity)"
+     *
+     * @return ADUC_FileEntity* Non-const address of wrapped object.
+     * @details Will uninitialize the current field as the assumption is the entity will be reinitialized by the caller.
+     */
+    ADUC_FileEntity* address_of()
+    {
+        uninit();
```

Comment:
what don't we just make uninit() public, and require caller to explicitly call uninit() before calling address_of() ?

Reply:
I see what you mean and it would make sense for const ptr.

I was aligning with similar pattern in STRING_HANDLE_wrapper for address_of() method: https://github.com/Azure/iot-hub-device-update/blob/feature/wiot-s1/src/utils/string_utils/inc/aduc/string_handle_wrapper.hpp#L56

I think if need a const pointer then can create a method for that later when needed; otherwise, the usage pattern is expecting that it will get overwritten so need to uninit.

Reply:
fixed with simplification of this file